### PR TITLE
Feature/1343 fixed width files hdfs

### LINKED ISSUE #1343: Allow fixed width on HDFS

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
Fixes #1343 

This PR is based on https://github.com/apache/metamodel/pull/105.

### CODE REVIEW

**Comment by LosD** [comment]
Travis error is because of -SNAPSHOT depencency

**Comment by LosD** [comment]
Will this keep compatibility with old configurations?

**Comment by ClaudiaPHI** [comment]
Cannot upgrade the home folder if there are fixed-width datastore saved in user preferences. I will investigate the issue. 
maybe @kaspersorensen knows why.

**Comment by ClaudiaPHI** [comment]
The upgrade of old datastores to new ones has been fixed, but I need to wait for metamodel fixedwidth project to be stable in order to integrate my changes to that project.

<b> NOT MERGEABLE</b>

**Comment by ClaudiaPHI** [comment]
Ready to merge.  Waiting for Metamodel release

**Comment by LosD** [comment]
Then I guess it's _not_ ready to merge? Master shouldn't depend on an external -SNAPSHOT release (since builds wouldn't be repeatable).

**Comment by LosD** [comment]
It's been waiting for months now, I guess we can wait a few days more :)

File: engine/xml-config/src/test/java/org/datacleaner/configuration/DomConfigurationWriterTest.java [hunk 1]
```diff
@@ -404,6 +405,44 @@ public void testWriteAndReadAllStringPatterns() throws Exception {
         assertEquals(false, regexStringPattern.isMatchEntireString());
     }
 
+    @Test
+    public void testWriteAndReadFixedWidthDatastore() throws Exception {
+        final FileResource fileResource = new FileResource("C://test.csv");
```
Comment (LosD) [inline engine/xml-config/src/test/java/org/datacleaner/configuration/DomConfigurationWriterTest.java]: Do not make assumptions about the file system. How would this test ever pass in a non-Windows environment?

File: engine/xml-config/src/test/resources/example-job-fixed-width-datastore.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<configuration xmlns="http://eobjects.org/analyzerbeans/configuration/1.0">
+  <datastore-catalog>
+<fixed-width-datastore description="bar" name="my fixed width ds">
+  <filename>C:\test.csv</filename>
```
Comment (LosD) [inline engine/xml-config/src/test/resources/example-job-fixed-width-datastore.xml]: Same here

**Comment by LosD** [comment]
-1

Test failures and whole-file reformats needs to go.

**Comment by LosD** [comment]
Something seems to assume that there is a Hadoop configuration loaded.

**Comment by ClaudiaPHI** [comment]
Do not merge. I really made a HUGE mistake with trying to merge squash the commits.

**Comment by ClaudiaPHI** [comment]
Closing as being duplicate of https://github.com/datacleaner/DataCleaner/pull/1489

**Pull request closed without merging**